docs/API.md

```diff
@@ -104,133 +184,191 @@ gulp.task('somename', function() {
 
 #### name
 
-The name of the task. Tasks that you want to run from the command line should not have spaces in them.
+Optional, The name of the task. Tasks that you want to run from the command line
+should not have spaces in them.
 
-#### deps
-Type: `Array`
+If the name is not provided, the task will be named after the function 
+`name` attribute, set on any named function.
 
-An array of tasks to be executed and completed before your task will run.
+[Function.name] is not writable; it cannot be set or edited.
+It will be empty for anonymous functions:
 
 ```js
-gulp.task('mytask', ['array', 'of', 'task', 'names'], function() {
-  // Do stuff
-});
+function foo() {};
+foo.name === 'foo' // true
+
+var bar = function() {};
+bar.name === '' // true
+
+bar.name = 'bar'
+bar.name === '' // true
 ```
 
-**Note:** Are your tasks running before the dependencies are complete?  Make sure your dependency tasks are correctly using the async run hints: take in a callback or return a promise or event stream.
+You should either provide the task name or avoid anonymous functions.
 
 #### fn
 
-The function that performs the task's operations. Generally this takes the form of `gulp.src().pipe(someplugin())`.
+The function that performs the task's operations. Generally this takes the 
+form of `return gulp.src().pipe(someplugin());`.
```

Comment:
Like:

    The function that performs the task's operations. Generally it takes this 
    form:
    ```
    gulp.task('someTask', function() {
      return gulp.src(['some/glob/**/*.ext']).pipe(someplugin());
    })
    ```